ConditionalCompilation.md

```diff
@@ -1,28 +1,139 @@
 # Conditional Compilation
 
-(TBD)
-
-This section will discuss extensions to WGSL
-to support conditional compilation, where the conditions
-are set by the host code runtime or build time.
-
-* `#if #else #endif` (or perhaps `#ifdef`) in the style of a C preprocessor 
-  is the approach that comes first to mind.
-* conditional compilation runs after simple templating but before import processing..
-  * (note that we may want to disable code that may not be correct so parsing first is iffy)
-* best to define default values in the wgsl for all conditionals
-  * IDE tools can't assume any definitions that aren't in the wgsl.
-* alternately, consider some kind of [cfg] on elements instead? 
-  * More structured: ties to source elements not lines
-    * Perhaps some optimization potential..
-      * For example could parse/lower certain functions ahead of 
-        time and easily remove them from the IR vs #if #else which operates on a source level.
-    * Is it an wgsl attribute? syntax makes sense but wgsl 
-* what are some examples of things to conditionally compile?
-  * function variants e.g. select which function variant to use based on condition
-  * struct fields e.g. include fields only if condition is set
-  * import statements? e.g. import util/fast vs util/regular based on condition
-* how do we support build time vs runtime linking?
-  * e.g. some projects will want prebuilt versions for various mobile configurations.
-    * ref [unity shader conditionals](https://docs.unity3d.com/Manual/shader-conditionals.html)
-  * should the wesl code distinguish conditions that are settable at compile time from those that are settable at runtime?
+## Overview
+
+> This section is non-normative
+
+Conditional compilation is a mechanism to modify the source code based on parameters passed to the compiler.
+This specificaton extends the [*attribute* syntax](https://www.w3.org/TR/WGSL/#attributes) with three new attributes: `@if`, `@elif`, `@else`.
+These attributes indicate that the syntax node they decorate can be removed by the compiler based on feature flags.
+
+> Note: This implementation is similar to the `#[cfg(feature = "")]` syntax in rust.
+
+### Usage Example
+
+```rs
```

Comment:
love the examples! 

(also, reminder to drop @else from example if dropping @else)